### DISCUSSION TITLE
High order thinking?

### DISCUSSION THREAD

**Discussion by BluePrintRandom** [lucidrains/imagen-pytorch#49] [Ideas]
say we give a seed image like a puzzle, and a solved maze as training data,

could pytorch / neural nets solve generalized puzzles?

could a NN outperform A*?

**Comment by jacobwjs** [comment]
Sounds like a problem better suited for reinforcement learning :)
